modules/core/shared/src/main/scala/lucuma/core/util/TimestampInterval.scala

```diff
@@ -85,12 +85,15 @@ sealed class TimestampInterval private (val start: Timestamp, val end: Timestamp
    */
   def minus(other: TimestampInterval): List[TimestampInterval] =
     overlap(other) match {
-      case Overlap.LowerPartial   => List(between(start, other.start))
-      case Overlap.UpperPartial   => List(between(other.end, end))
-      case Overlap.ProperSuperset => List(between(start, other.start), between(other.end, end))
-      case Overlap.None           => List(this)
+      case Overlap.LowerPartial                                  => List(between(start, other.start))
+      case Overlap.UpperPartial                                  => List(between(other.end, end))
+      case Overlap.ProperSuperset if (other.start === other.end) => List(this)
+      case Overlap.ProperSuperset if (start === other.start)     => List(between(other.end, end))
+      case Overlap.ProperSuperset if (end === other.end)         => List(between(start, other.start))
+      case Overlap.ProperSuperset                                => List(between(start, other.start), between(other.end, end))
+      case Overlap.None                                          => List(this)
       case Overlap.ProperSubset |
-           Overlap.Equal          => Nil
+           Overlap.Equal                                         => Nil
```

Comment:
Hmm, I wonder if this should produce a `TimestampUnion` instead of a `List[TimestampInterval]`.  In the original code when the `List` is added to the union it gets merged in as expected.

Reply:
So the disjoint "empty" intervals didn't cause a problem? It seems like that would create a Union with 2 pieces. But, maybe the empty piece is expected?

Reply:
Expected by your code, that is. Mine doesn't expect it. But, I could filter it out.

Reply:
An empty interval would add nothing to the sum, so in that sense it doesn't matter for time accounting.